namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala

```diff
@@ -316,31 +316,33 @@ class HttpControlService(storage: DtabStore, delegate: Ns => NameInterpreter, na
     }
 
   private[this] val bindingCacheMu = new {}
-  private[this] var bindingCache: Map[(String, Path), Activity[NameTree[Name.Bound]]] = Map.empty
-  private[this] def getBind(ns: String, path: Path): Activity[NameTree[Name.Bound]] =
+  private[this] var bindingCache: Map[(String, Path, String), Activity[NameTree[Name.Bound]]] = Map.empty
```

Comment:
I don't think we want to key off of the extra dtab in the cache.  What do you think about making extraDtab a `Option[String]` in `getBind`?  If the extraDtab is present, then skip the cache and just return `delegate(ns).bind(...)` directly.  If it's absent, then use the cache like before.

Does that seem reasonable?